shared/src/bad_token/trace_call.rs

```diff
@@ -369,6 +380,30 @@ mod tests {
                 state_diff: None,
                 transaction_hash: None,
             },
+            BlockTrace {
+                output: Default::default(),
+                trace: Some(vec![TransactionTrace {
+                    trace_address: Vec::new(),
+                    subtraces: 0,
+                    action: Action::Call(Call {
+                        from: H160::zero(),
+                        to: H160::zero(),
+                        value: 0.into(),
+                        gas: 0.into(),
+                        input: Bytes(Vec::new()),
+                        call_type: CallType::None,
+                    }),
+                    action_type: ActionType::Call,
+                    result: Some(Res::Call(CallResult {
+                        gas_used: 1.into(),
+                        output: Bytes(Vec::new()),
+                    })),
+                    error: None,
+                }]),
+                vm_trace: None,
+                state_diff: None,
+                transaction_hash: None,
+            },
```

Comment:
Maybe I am missing something, but which part of this `BlockTrace` indicated that the simulation failed on infinite token approval?

Reply:
That test is called handle_response_ok and as such does not test the error case. The error case test is the #ignored mainnet one.